docs/design/kotlin-smithy-sdk.md

```diff
@@ -150,60 +141,47 @@ class MyStruct private constructor(builder: BuilderImpl) {
         return "MyStruct(foo=$foo, bar=$bar, baz=$baz, yesno=$yesno)"
     }
 
-    fun toBuilder(): Builder = BuilderImpl(this)
+    fun toBuilder(): Builder = Builder(this)
 
     fun copy(foo: String? = this.foo,
              bar: Int = this.bar,
              baz: Baz? = this.baz,
              yesno: SimpleYesNo? = this.yesno): MyStruct {
-        val builder = BuilderImpl(this)
+        val builder = Builder(this)
         builder.foo = foo
         builder.bar = bar
         builder.baz = baz
         builder.yesNoAsString = yesno?.toString()
         return builder.build()
     }
 
-    fun copy(block: DslBuilder.() -> Unit = {}): MyStruct = BuilderImpl(this).apply(block).build()
+    fun copy(block: Builder.() -> Unit = {}): MyStruct = Builder(this).apply(block).build()
```

Comment:
Question/suggestion

This should also probably be `inline` with concrete builder type being public